src/pages/index.js

```diff
@@ -12,54 +11,140 @@ import {
   profileEditForm,
   addCardButton,
   addCardForm,
+  changeAvatarButton,
+  changeAvatarForm,
 } from "../utils/constants.js";
 import Card from "../components/Card.js";
 import FormValidator from "../components/FormValidator.js";
 import ModalWithForm from "../components/ModalWithForm.js";
 import ModalWithImage from "../components/ModalWithImage.js";
 import UserInfo from "../components/UserInfo.js";
 import Section from "../components/Section.js";
+import Api from "../components/API.js";
+
+//API
+const api = new Api({
+  baseURL: "https://around-api.en.tripleten-services.com/v1",
+  authToken: "ea161ff4-cda4-4a20-a8db-64fec38336d8",
+});
+
+api
+  .getAppInfo()
+  .then(([userData, cards]) => {
+    user.setUserInfo({
+      name: userData.name,
+      bio: userData.about,
+    });
+    user.changeAvatarImage(userData.avatar);
+    section.renderItems(cards);
+  })
+  .catch((err) => {
+    console.error(`Failed to load app info: ${err}`);
+  });
 
 //instantiate
 const section = new Section(
   {
-    items: initialCards,
     renderer: (data) => {
       section.addItem(createCard(data));
     },
   },
   cardSelectors.cardListEl,
 );
 
-const newCardModal = new ModalWithForm(
-  "#add-card-modal",
-  handleAddCardFormSubmit,
-);
+const newCardModal = new ModalWithForm({
+  modalSelector: "#add-card-modal",
+  handleFormSubmit: (data) => {
+    newCardModal.renderLoading(true);
+    api
+      .addCard({
+        name: data.title,
+        link: data.link,
+      })
+      .then((data) => {
+        section.addItem(createCard(data));
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        newCardModal.renderLoading(false);
+        newCardModal.close();
+      });
+  },
+});
 
-const editProfileModal = new ModalWithForm(
-  "#profile-edit-modal",
-  handleProfileEditFormSubmit,
-);
+const editProfileModal = new ModalWithForm({
+  modalSelector: "#profile-edit-modal",
+  handleFormSubmit: (data) => {
+    editProfileModal.renderLoading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```